workspaces/optic-engine/src/interactions/visitors/diff.rs

```diff
@@ -125,7 +125,7 @@ impl QueryParametersVisitor<InteractionDiffResult> for DiffQueryParametersVisito
       query_shape_id,
     ) {
       (None, None, None) => {}
```

Comment:
It's this line that's preventing us from always representing the query parameters as an empty object when parsing it from the interaction: if no query params are specced and non are sent, we don't want to yield any diffs. With the default fallback, that would generate empty query parameters diffs for every existing endpoint for every spec out there, even if no query parameters are used.

So instead, we only consider the query parameters an object empty when they are not sent in the contexts where this is helpful. So far, that's only in shape diffing them. In that context, query parameters have definitely been specced, not sending any doesn't mean we just get to ignore the spec!